### ISSUE TITLE
[packagers] asdf.repository does not inherit name from parent packager

### ISSUE THREAD

**Issue by aalmiray** [jreleaser/jreleaser#1592]
### Task List

- [x] Steps to reproduce provided
- [ ] Stacktrace (if present) provided
- [x] Example that reproduces the problem (link to git repository is ideal)
- [x] Full description of the issue provided (see below)

### Steps to Reproduce

1. clone https://github.com/jreleaser/helloworld-java-bin
2. edit `jreleaser.yml` and add the following
```yaml
packagers:
  asdf:
    active: ALWAYS
    repository:
      owner: aalmiray
      username: aalmiray
      name: asdf-test
```
3. execute `jreleaser config` and inspect the result

```
packagers:
    asdf:
        enabled: true
        active: ALWAYS
        continueOnError: false
        commitAuthor:
            name: jreleaserbot
            email: jreleaser@kordamp.org
        repository:
            enabled: false
            owner: aalmiray
            name: asdf-test
            tagName: v{{projectVersion}}
            branch: HEAD
            branchPush: HEAD
            username: jreleaser
            token: ************
 
distributions:
    helloworld:
        enabled: true
        active: ALWAYS
        type: JAVA_BINARY
        executable:
            name: helloworld
            windowsExtension: bat
        artifacts:
            artifact 0:
                enabled: true
                active: ALWAYS
                path: /private/tmp/helloworld-java-bin/out/jreleaser/assemble/helloworld/java-archive/helloworld-1.0.0.zip
        stereotype: CLI
        java:
            enabled: true
            version: 11
            groupId: test.jreleaser
            artifactId: helloworld
            mainClass: org.jreleaser.examples.HelloWorld
            multiProject: false
        asdf:
            enabled: true
            active: ALWAYS
            continueOnError: false
            templateDirectory: src/jreleaser/distributions/helloworld/asdf
            commitAuthor:
                name: jreleaserbot
                email: jreleaser@kordamp.org
            toolcheck: {{distributionExecutable}} --version
            repository:
                enabled: true
                active: RELEASE
                owner: aalmiray
                name: asdf-helloworld
                tagName: v{{projectVersion}}
                branch: HEAD
                branchPush: HEAD
                username: jreleaser
                token: ************
                commitMessage: {{distributionName}} {{tagName}}
```
### Expected Behaviour

The asdf section at the helloworld distribution shows `asdf-helloworld` as the resolved name

### Actual Behaviour

Resolved name should be `asdf-test` as that is the value set in the parent packager.

### Environment Information

- **Operating System**: OSX
- **JReleaser Version:** 1.10.0
- **JDK Version:** 17

**Comment by aalmiray** [comment]
These packagers are also affected:
- appimage
- jib
- winget

**Comment by aalmiray** [comment]
🎉 This issue has been resolved in `v1.11.0` ([Release Notes](https://github.com/jreleaser/jreleaser/releases/tag/v1.11.0))
